Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM golang:1.21-alpine
+
+RUN apk update && apk add --no-cache git
+
+WORKDIR /app
+
+# Copy go.mod and go.sum first
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy the rest of the code
+COPY . .
+
+# Tidy up Go modules before building
+RUN go mod tidy
+
+# Build the app
+RUN go build -o main .
```

Comment:
I think the built binary should have a specific name not just be called main. It should also remove any versioning etc using `-ldflags="-s -w"` in the build command